pipelines/salesforce/__init__.py

```diff
@@ -10,133 +10,152 @@
-    """
-    return True
-
-def get_records(
-    sf: Salesforce,
-    sobject: str,
-    last_state:Optional[str] = None,
-    replication_key:Optional[str] = None,
-) -> Iterable[Dict[str, Any]]:
-
-    # Get all fields for the sobject
-    desc = getattr(sf, sobject).describe()
-    # Salesforce returns compound fields as separate fields, so we need to filter them out
-    compound_fields = {
-        f["compoundFieldName"]
-        for f in desc["fields"]
-        if f["compoundFieldName"] is not None
-    } - {"Name"}
-    # Salesforce returns datetime fields as timestamps, so we need to convert them
-    date_fields = {
-        f["name"] for f in desc["fields"] if f["type"] in ("datetime",) and f["name"]
-    }
-    # If no fields are specified, use all fields except compound fields
-    fields = [f["name"] for f in desc["fields"] if f["name"] not in compound_fields]
-
-    # Generate a predicate to filter records by the replication key
-    predicate, order_by, n_records = "", "", 0
-    if replication_key:
-        if last_state:
-            predicate = f"WHERE {replication_key} > {last_state}"
-        order_by = f"ORDER BY {replication_key} ASC"
-    query = f"SELECT {', '.join(fields)} FROM {sobject} {predicate} {order_by}"
-    if not is_production():
-        query += " LIMIT 100"
-
-    # Query all records in batches
-    for page in getattr(sf.bulk, sobject).query_all(query, lazy_operation=True):
-        for record in page:
-            # Strip out the attributes field
-            record.pop("attributes", None)
-            for field in date_fields:
-                # Convert Salesforce timestamps to ISO 8601
-                if record.get(field):
-                    record[field] = pendulum.from_timestamp(
-                        record[field] / 1000,
-                    ).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
-        yield page
-        n_records += len(page)
 
 @dlt.source(name="salesforce")
 def salesforce_source(
     user_name: str = dlt.secrets.value,
     password: str = dlt.secrets.value,
     security_token: str = dlt.secrets.value,
-) ->Iterable[DltResource]:
+) -> Iterable[DltResource]:
+    """
+    Retrieves data from Salesforce using the Salesforce API.
+
+    Args:
+        user_name (str): The username for authentication. Defaults to the value in the `dlt.secrets` object.
+        password (str): The password for authentication. Defaults to the value in the `dlt.secrets` object.
+        security_token (str): The security token for authentication. Defaults to the value in the `dlt.secrets` object.
+
+    Yields:
+        DltResource: Data resources from Salesforce.
+    """
+
     client = Salesforce(user_name, password, security_token)
 
     # define resources
     @dlt.resource(write_disposition="replace")
-    def sf_user() -> Iterator[Dict[str, Any]]:
-        yield from get_records(client, "User")
+    def sf_user() -> Iterable[TDataItem]:
+        yield get_records(client, "User")
```

Comment:
I have a suspicion that dropping `from` in `yield from` results in function that yields a generator (not a value) since `get_records()` is a generator itself . Correct me if I'm wrong:

```python
>>> def get_records():
...     for i in range(10):
...         yield i
...
>>> def sf_user():
...     yield get_records()
...
>>> sf_user()
<generator object sf_user at 0x109833ac0>
>>> next(sf_user())
<generator object get_records at 0x1098f4110>
```

Reply:
the from part moved in to get_records. The tests pass too, so I think it should be fine